### PR TITLE
feat(ci): add a sample of auto testing codes

### PR DESCRIPTION
**Pull request by takeutak**
This PR is related with Issue #443  .

Signed-off-by: Takuma TAKEUCHI takeuchi.takuma@fujitsu.com

### CODE REVIEW

**Comment by takeutak** [comment]
Contents:
- Add a new sample for automatic code testing according to Issue #443 .

The following README describes their boot method and how to use them:
- https://github.com/takeutak/cactus/blob/feat_ci_add_a_sample_of_auto-testing_codes/packages/manual-of-auto-testing-codes.md

File: packages/routing-interface/util/RIFUtil.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export class RIFUtil {
             return JSON.parse(jsonStr);
         }
         catch (error) {
-            logger.error(error)
+            // logger.error(error)
```
Comment (petermetz) [inline packages/routing-interface/util/RIFUtil.ts:46]: @takeutak Is this change intended to be in #444 or #442 ?

cc: @sfuji822 @hartm @jonathan-m-hamilton

File: packages/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
+/*
```
Comment (petermetz) [inline packages/jest.config.js:1]: @takeutak Is this change intended to be in #444 or #442 ?

cc: @sfuji822 @hartm @jonathan-m-hamilton

**Review by petermetz — COMMENTED** [review]
See questions inline

File: packages/jest.config.js (hunk 2 above)
Comment (petermetz) [reply packages/jest.config.js:1]: Clarified over the maintainer's call: #442

File: packages/routing-interface/util/RIFUtil.ts (hunk 1 above)
Comment (petermetz) [reply packages/routing-interface/util/RIFUtil.ts:46]: Clarified over the maintainer's call: #442

Comment (takeutak) [reply packages/routing-interface/util/RIFUtil.ts]: Thank you for your comment.  As your comment on PR #442, I corrected this file.
https://github.com/hyperledger/cactus/pull/444/commits/b2d1f476f626122cdeaaa68aaeb08099fa496e41

File: packages/jest.config.js (hunk 2 above)
Comment (takeutak) [reply packages/jest.config.js:1]: Thank you for your comment.
As I said on the maintainer call, jest.config.js should be included by PR #444 (not #442).  So I deleted this file on PR #442.

Comment (takeutak) [reply packages/jest.config.js:1]: Moreover, I answered your question on PR #442.

> petermetz 3 days ago Member
> @takeutak Is this file scoped for the entire project (e.g. all packages) or just a sub-set of packages?

Just a sub-set of packages.
Please refer the line 123 on jest.config.js. The range is defined as the following.
roots: [
  "<rootDir>/ledger-plugin",
  "<rootDir>/routing-interface"
],

https://github.com/hyperledger/cactus/pull/442/files/0a2d04f0d30f25791c554a57412a188dfc7a670e#diff-bdf78ae09a2d9e94fa7baf7fa0103c048873338937c22c1d89f4786b214e01deR123

**Comment by takeutak** [comment]
@petermetz I answered to your questions as the above.  Would you mind if you confirm this pull-request again?

File: packages/jest.config.js (hunk 2 above)
Comment (petermetz) [reply packages/jest.config.js:1]: > Moreover, I answered your question on PR #442.
> 
> > petermetz 3 days ago Member
> > @takeutak Is this file scoped for the entire project (e.g. all packages) or just a sub-set of packages?
> 
> Just a sub-set of packages.
> Please refer the line 123 on jest.config.js. The range is defined as the following.
> 
> ```
> roots: [
>  "<rootDir>/ledger-plugin",
>  "<rootDir>/routing-interface"
> ],
> ```
> 
> https://github.com/hyperledger/cactus/pull/442/files/0a2d04f0d30f25791c554a57412a188dfc7a670e#diff-bdf78ae09a2d9e94fa7baf7fa0103c048873338937c22c1d89f4786b214e01deR123

@takeutak Thank you for the info! Follow-up questions: 
Do you intend to have these tests executed as part of the CI in the future?
If yes to the above, how would the coverage statistics work? Is there a way to aggregate the coverage statistics between this newly introduced test runner and the pre-existing one?

Comment (petermetz) [reply packages/jest.config.js:1]: Discussed offline, we'll merge everything to support the TAP protocol.
https://testanything.org/

cc: @sfuji822 @hartm @jonathan-m-hamilton @takeutak

**Review by petermetz — APPROVED** [review]
LGTM

**Review by jonathan-m-hamilton — APPROVED** [review]
LGTM
